keras_cv/layers/feature_pyramid.py

```diff
@@ -0,0 +1,208 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+
+
+# TODO(scottzhu): Register it later due to the conflict in the retina_net
+# @tf.keras.utils.register_keras_serializable(package="keras_cv")
+class FeaturePyramid(tf.keras.layers.Layer):
+    """Implements a Feature Pyramid Network.
+
+    This implements the paper:
+      Tsung-Yi Lin, Piotr Dollar, Ross Girshick, Kaiming He, Bharath Hariharan, and
+      Serge Belongie.
+      Feature Pyramid Networks for Object Detection.
+      (https://arxiv.org/pdf/1612.03144)
+
+    Feature Pyramid Networks (FPNs) are basic components that are added to an
+    existing CNN to extract features at different scales. For the basic FPN, the
+    inputs are features `Ci` from different levels of a CNN, where the feature is
+    scaled from the image size by `1/2^i` for any level `i`.
```

Comment:
where the feature map size is scaled from the image size by a factor of ...

Reply:
Done.